packages/network/src/chart/Network.jsx

```diff
@@ -73,16 +86,64 @@ class Network extends React.PureComponent {
   }
 
   copyGraph(graph) {
-    const nodes = graph.nodes.map((node, index) => ({
-      ...node,
+    const { width, height, margin } = this.props;
+    const range = graph.nodes.reduce(
+      (minMax, node) =>
+        ({
+          x: {
+            min: Math.min(minMax.x.min, node.x),
+            max: Math.max(minMax.x.max, node.x),
+          },
+          y: {
+            min: Math.min(minMax.y.min, node.y),
+            max: Math.max(minMax.y.max, node.y),
+          },
+        }),
+      {
+        x: {
+          min: 999999,
+          max: -999999,
+        },
+        y: {
+          min: 999999,
+          max: -999999,
+        },
+      });
+
+    const actualWidth = width - margin.left - margin.right;
+    const actualheight = height - margin.top - margin.bottom;
+    const dataXRange = range.x.max - range.x.min;
+    const dataYRange = range.y.max - range.y.min;
+
+    const zoomLevel = Math.max(
+      dataXRange / actualWidth,
+      dataYRange / actualheight,
+    );
+
+    function xScale(x) {
+      const offset = margin.left +
+      (((actualWidth - (dataXRange / zoomLevel)) / 2) - (range.x.min / zoomLevel));
```

Comment:
you could potentially separate these calculations out into separate variables with meaningful names to make this more readable

Reply:
also, can you move (x and y) `offset`s out of the functions so you only have to compute them once?